picard/pluginmanager.py

```diff
@@ -449,26 +482,55 @@ def enabled(self, name):
-        if at least one needs to be updated, or False if no update is needed"""
+        """Compare available plugins versions with installed plugins ones
+        and yield plugin names of plugins that have new versions"""
         if self.available_plugins is not None:
             available_versions = {p.module_name: p.version for p in self.available_plugins}
             for plugin in self.plugins:
                 if plugin.module_name not in available_versions:
                     continue
                 if available_versions[plugin.module_name] > plugin.version:
-                    return True
-        return False
+                    yield plugin.name
 
     def check_update(self, parent=None):
         def _display_update():
-            if self._plugins_have_new_versions():
-                QMessageBox.information(
-                    parent,
-                    _("Picard Plugins Update"),
-                    _("There are updates available for your currently installed plugins."),
-                    QMessageBox.StandardButton.Ok, QMessageBox.StandardButton.Ok
-                )
+            UPDATE_LINES_TO_SHOW = 3
+            plugins_with_updates = sorted(self._plugins_have_new_versions())
+            if plugins_with_updates:
+                file_count = len(plugins_with_updates)
+                header = '<p>' + ngettext(
+                    "There is an update available for one of your currently installed plugins:",
+                    "There are updates available for your currently installed plugins:",
+                    file_count
+                ) + '</p>'
+                footer = '<p>' + ngettext(
+                    "Do you want to update the plugin now?",
+                    "Do you want to update the plugins now?",
+                    file_count
+                ) + '</p>'
+
+                extra_file_count = file_count - UPDATE_LINES_TO_SHOW
+                if extra_file_count > 0:
+                    extra_plugins = '<p>' + ngettext(
+                        "plus {extra_file_count:,d} other plugin.",
+                        "plus {extra_file_count:,d} other plugins.",
+                        extra_file_count).format(extra_file_count=extra_file_count) + '</p>'
+                else:
+                    extra_plugins = ''
+
+                plugin_list = ''
+                for plugin_name in plugins_with_updates[:UPDATE_LINES_TO_SHOW]:
+                    plugin_list += f"<li>{plugin_name}</li>"
+
+                msg = PluginUpdatesDialog(parent, f'{header}<ul>{plugin_list}</ul>{extra_plugins}{footer}')
+
+                show_options_page, perform_check = msg.show()
+                if parent:
+                    config = get_config()
+                    config.setting['check_for_plugin_updates'] = perform_check
+
+                    if show_options_page:
+                        parent.show_plugins_options_page()
```

Comment:
Should this be changed to:
```suggestion
                if parent:
                config = get_config()
                config.setting['check_for_plugin_updates'] = perform_check

                if parent and show_options_page:
                    parent.show_plugins_options_page()
```

Reply:
Changes tested and implemented.

Reply:
Yes I think so. The setting doesn't depend on `parent`.